cfn-resources/cloud-backup-restore-jobs/cmd/resource/resource.go

```diff
@@ -44,7 +45,7 @@ func Create(req handler.Request, prevModel *Model, currentModel *Model) (handler
 	}
 	// Create atlas client
 	if currentModel.Profile == nil || *currentModel.Profile == "" {
```

Comment:
As `util::NewMongoDBClient` is doing the null check for the profile, I think you can now remove it in all occurrences here

Reply:
Unfortunately, we cannot. In addition to the null check, we are also setting `currentModel.Profile = aws.String(util.DefaultProfile)` which is super important as without this assignment, the CFN stack would return the error "cfn resource is not returning the primary identifier (profile)"

Reply:
[q - for my own knowledge] Is passing a reference to `currentModel` into `util::NewMongoDBClient` an antipattern in golang? Just wondering can/should we do it all through the helper method.

Reply:
`currentModel` is different for every CFN resource 😞 . I was thinking to pass `interface{}` to  `NewMongoDBClient` but I haven't found a way to set the field `Profile` in the interface when the user does not provide the profile name.

Reply:
Got it - yes the interface approach is interesting, but I'd need to play with it a bit to see could it work as well.

Reply:
Comment isn't blocking, just ideation